### PR TITLE
Update event handling docs.

### PR DESCRIPTION
**Pull request by anntzer**
Much reformatting.

Remove documentation for `picker=<float>` which is deprecated in favor
of `pickradius`.

Add reference to default keymaps (closes https://github.com/matplotlib/matplotlib/issues/11472).

Modernize examples: use `pickradius`, use `subplots()` instead of
`add_subplot(111)`, etc.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: doc/users/event_handling.rst [hunk 1]
```diff
@@ -138,22 +127,19 @@ is created every time a mouse is pressed::
...
-:class:`~matplotlib.backend_bases.LocationEvent`, so we have access to
-the data and pixel coordinates in event.x and event.xdata.  In
-addition to the ``LocationEvent`` attributes, it has
+The `.MouseEvent` that we just used is a `.LocationEvent`, so we have access to
+the data and pixel coordinates via ``(event.x, event.y)`` and ``(event.xdata,
+event.ydata)``.  In addition to the ``LocationEvent`` attributes, it also has
 
     ``button``
-        button pressed None, 1, 2, 3, 'up', 'down' (up and down are used for scroll events)
+        the button pressed: None, 1, 2, 3, 'up', 'down' (up and down are used for scroll events)
```
Comment (timhoffm) [inline doc/users/event_handling.rst:142]:
```suggestion
        the button pressed: None or `.MouseButton`
```

File: doc/users/event_handling.rst [hunk 2]
```diff
@@ -162,12 +148,12 @@ Draggable rectangle exercise
 ----------------------------
 
 Write draggable rectangle class that is initialized with a
-:class:`~matplotlib.patches.Rectangle` instance but will move its x,y
+`.Rectangle` instance but will move its x,y
```
Comment (timhoffm) [inline doc/users/event_handling.rst:151]:
```suggestion
`.Rectangle` instance but will move its ``xy``
```

File: doc/users/event_handling.rst [hunk 3]
```diff
@@ -194,43 +180,41 @@ Here is the solution::
...
+            if event.inaxes != self.rect.axes:
+                return
             contains, attrd = self.rect.contains(event)
-            if not contains: return
+            if not contains:
+                return
             print('event contains', self.rect.xy)
-            x0, y0 = self.rect.xy
-            self.press = x0, y0, event.xdata, event.ydata
+            self.press = self.rect.xy, event.xdata, event.ydata
```
Comment (timhoffm) [inline doc/users/event_handling.rst:190]: This would be `(x0, y0), event.xdata, event.ydata`. Is this change intentional?

Edit: Seems like. But then I'd go for `self.rect.xy, (event.xdata, event.ydata)` so that below you have `(x0, y0), (xpress, ypress) = self.press`. Using a tuple only for one of the coordinates would be a bit awkward.

File: doc/users/event_handling.rst [hunk 4]
```diff
@@ -267,14 +252,15 @@ Extra credit solution::
...
+                    or DraggableRectangle.lock is not None):
+                return
             contains, attrd = self.rect.contains(event)
-            if not contains: return
+            if not contains:
+                return
             print('event contains', self.rect.xy)
-            x0, y0 = self.rect.xy
-            self.press = x0, y0, event.xdata, event.ydata
+            self.press = self.rect.xy, event.xdata, event.ydata
```
Comment (timhoffm) [inline doc/users/event_handling.rst:263]: See above.

File: doc/users/event_handling.rst [hunk 5]
```diff
@@ -403,100 +383,77 @@ background that the mouse is over::
...
+    ``callable``
+        If picker is a callable, it is a user supplied function which
+        determines whether the artist is hit by the mouse event.  The
+        signature is ``hit, props = picker(artist, mouseevent)`` to
+        determine the hit test.  If the mouse event is over the artist,
+        return ``hit = True``; ``props`` is a dictionary of properties that
+        become additional attributes on the `.PickEvent`.
+
+The artist's ``pickradius`` property can additionally be set to a tolerance
+value (in points; there are 72 points per inch) that determines how far the
```
Comment (timhoffm) [inline doc/users/event_handling.rst:410]:
```suggestion
value in points (there are 72 points per inch) that determines how far the
```
IMHO the unit is first-class information that should not be parenthesed.

If you think that's too ambiguous, use e.g. `tolerance value in units of points`

File: doc/users/event_handling.rst [hunk 6]
```diff
@@ -526,41 +482,39 @@ can use multiple subplots to plot the multiple time series.
...
+        n = len(event.ind)
+        if not n:
+            return
+        fig, axs = plt.subplots(n, squeeze=False)
+        for dataind, ax in zip(event.ind, axs.flat):
             ax.plot(X[dataind])
-            ax.text(0.05, 0.9, 'mu=%1.3f\nsigma=%1.3f'%(xs[dataind], ys[dataind]),
-                    transform=ax.transAxes, va='top')
+            ax.text(0.05, 0.9,
+                    'mu=%1.3f\nsigma=%1.3f' % (xs[dataind], ys[dataind]),
```
Comment (timhoffm) [inline doc/users/event_handling.rst:512]: Fancy using TeX?

```suggestion
                    f'$\mu$={xs[dataind]:1.3f}\n$\sigma$={ys[dataind]:1.3f}',
```

**Review by timhoffm — COMMENTED** [review]
On a general note (not part of this PR). Should this be a tutorial?

**Comment by anntzer** [comment]
yes to all (except for turning into a tutorial, which can be done separately)

**Review by timhoffm — APPROVED** [review]
